pandas/core/groupby.py

```diff
@@ -1023,6 +1024,61 @@ def tail(self, n=5):
         tail = obj[in_tail]
         return tail
 
+    def pipe(self, func, *args, **kwargs):
+        """
+        Apply func(self, \*args, \*\*kwargs)
+
+        .. versionadded:: 0.16.3
+
+        Parameters
+        ----------
+        func : function
+        function to apply to the %(klass)s.
+        ``args``, and ``kwargs`` are passed into ``func``.
+        Alternatively a ``(callable, data_keyword)`` tuple where
+        ``data_keyword`` is a string indicating the keyword of
+        ``callable`` that expects the %(klass)s.
+        args : positional arguments passed into ``func``.
+        kwargs : a dictionary of keyword arguments passed into ``func``.
+
+        Returns
+        -------
+        object : the return type of ``func``.
+
+        Notes
+        -----
+
+        Use ``.pipe`` when chaining together functions that expect
+        on Series or DataFrames, or GroupBys. Instead of writing
+
+        >>> f(g(h(df), arg1=a), arg2=b, arg3=c)
+
+        You can write
+
+        >>> (df.pipe(h)
+        ...    .pipe(g, arg1=a)
```

Comment:
this example should be relevant to a groupby. Pls test make an explicit test for the example as well.